File: Winter2022/examples/java/Lecture 2/MusicPlayingServiceExample/app/src/main/java/course/examples/services/musicservice/MusicServiceClient.java
package course.examples.services.musicservice;

import android.app.Activity;
import android.content.Intent;
import android.os.Bundle;
import android.widget.Button;

public class MusicServiceClient extends Activity {

	@Override
	public void onCreate(Bundle savedInstanceState) {
		super.onCreate(savedInstanceState);

		setContentView(R.layout.main);
		
		// Intent used for starting the MusicService
		final Intent musicServiceIntent = new Intent(getApplicationContext(),
				MusicService.class);

		final Button startButton = findViewById(R.id.start_button);
		startButton.setOnClickListener(src -> {
			// Start the MusicService using the Intent
			startService(musicServiceIntent);
		});

		final Button stopButton = findViewById(R.id.stop_button);
		stopButton.setOnClickListener(src -> {
			// Stop the MusicService using the Intent
			stopService(musicServiceIntent);
		});
	}
}

File: Fall2022/examples/Lecture 2/MusicPlayingServiceExample/app/src/main/java/course/examples/services/musicservice/MusicServiceClient.kt
package course.examples.services.musicservice

import android.app.Activity
import android.os.Bundle
import android.content.Intent
import android.view.View
import android.widget.Button

class MusicServiceClient : Activity() {
    public override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.main)

        // Intent used for starting the MusicService
        val musicServiceIntent = Intent(
            applicationContext,
            MusicService::class.java
        )
        val startButton = findViewById<View>(R.id.start_button) as Button
        startButton.setOnClickListener { // Start the MusicService using the Intent
            startService(musicServiceIntent)
        }
        val stopButton = findViewById<View>(R.id.stop_button) as Button
        stopButton.setOnClickListener { // Stop the MusicService using the Intent
            stopService(musicServiceIntent)
        }
    }
}